.github/workflows/version_compatability.yml

```diff
@@ -0,0 +1,44 @@
+# GitHub Actions for genCMPClient
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) Siemens AG, 2021-2024
+
+name: OpenSSL compatabilty
+
+on: push
+
+jobs:
+  openssl_version:
+    strategy:
+      fail-fast: false
+      matrix:
+        branch: [openssl-3.0, openssl-3.1, openssl-3.2, openssl-3.3, openssl-3.4]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: build openssl
+        run: |
+          git clone https://github.com/openssl/openssl.git --branch ${{ matrix.branch }} --depth 1
+          cd openssl
+          ./config --banner=Configured --strict-warnings --prefix=/usr/local/${{ matrix.branch }} --openssldir=/usr/local/${{ matrix.branch }}/ssl shared zlib
+          make -s -j4
+          sudo make install_sw install_ssldirs
+          cd ..
+          echo "OPENSSL_DIR=./openssl" >> $GITHUB_ENV
+      - name: make
+        run: |
+          make -f Makefile_v1
+          make -f Makefile_v1 clean_all
+          USE_LIBCMP=1 STATIC_LIBCMP=1 make -f Makefile_v1 build_no_tls
```

Comment:
please add: 
```
make -f Makefile_v1 test_Mock
```

Reply:
I believe the current build failure is due to consistency of the build configuration after 
`USE_LIBCMP=1 STATIC_LIBCMP=1 make -f Makefile_v1 build_no_tls`.

So instead of adding `make -f Makefile_v1 test_Mock` thereafter,
one could replace the earlier `make -f Makefile_v1` by `make -f Makefile_v1 test_Mock`.
or insert `make -f Makefile_v1 clean_all` before the mock test build,
of more efficiently: 
augment the new `make -f Makefile_v1 test_Mock` by
`USE_LIBCMP=1 STATIC_LIBCMP=1 make -f Makefile_v1 test_Mock`